src/components/RepoBadge/RepoBadge.js

```diff
@@ -0,0 +1,73 @@
+import React, { useMemo } from 'react'
+import { blockExplorerUrl, Link } from '@aragon/ui'
+import { network } from '../../environment'
+import AppIcon from '../../components/AppIcon/AppIcon'
+import { KNOWN_ICONS, isKnownRepo } from '../../repo-utils'
+import { repoBaseUrl, parseHub } from '../../url-utils'
+
+function useRepoBadge(repo) {
+  const {
+    content: { appId, contractAddresses, name, icons },
+    version,
+  } = repo
+
+  return {
+    baseUrl: repoBaseUrl(repo),
+    icons: icons,
+    contractAddress: contractAddresses,
+    knownIcon: KNOWN_ICONS.has(appId) ? KNOWN_ICONS.get(appId) : null,
+    name: name,
+  }
+}
+
+const RepoBadge = ({ repo }) => {
+  const { baseUrl, icons, contractAddress, knownIcon, name } = useRepoBadge(
```

Comment:
Rather than defining a new hook for this, we can just memoize it

Reply:
Yeah, this took me a while, I kept thinking about it and in the end decided to go with the hook to separate any logic (even just destructuring) with the view, I'd like to keep the separation if that's alright with you and if it makes sense also add the `memoization`

Reply:
I don't think `memoization` is needed though as these are in sync

Reply:
Sure... I just don't really see the point in adding an extra hook just to do this though, especially if the hook is going to be rerun on every render anyway.

> I don't think memoization is needed though as these are in sync

I'm not quite sure what you meant by this, but we could avoid re-renders (not that we necessarily want / need to) if we applied `React.memo()` to the entire component, used a `useMemo()` in the hook, or used a `useMemo()` in the component and inlined the logic.